roofit/roostats/inc/RooStats/MCMCCalculator.h

```diff
@@ -195,12 +195,8 @@ namespace RooStats {
       void SetupBasicUsage();
       void SetBins(const RooAbsCollection& coll, Int_t numBins) const
       {
-         TIterator* it = coll.createIterator();
-         RooAbsArg* r;
-         while ((r = (RooAbsArg*)it->Next()) != NULL)
-            if (dynamic_cast<RooRealVar*>(r))
-               ((RooRealVar*)r)->setBins(numBins);
-         delete it;
+         for (auto *r : static_range_cast<RooRealVar *>(coll))
+               r->setBins(numBins);
```

Comment:
```suggestion
         for (auto *r : dynamic_range_cast<RooRealVar *>(coll)) {
            if (r) {
               r->setBins(numBins);
            }
          }
```
If there was a `dynamic_cast` where the result is explicitly checked for `nullptr`, there is probably a good reason for this.